### PR TITLE
Deployment step for disabling redemptions on mainnet

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Depends on: #446 
Closes: #437

For the initial release based on optimistic minting, the off-chain client will not have the functionality allowing to perform redemptions. For this reason, we need to temporarily disable redemptions in the `Bridge` contract. We achieve that by adding an additional step to the deployment scripts. This step is invoked for `mainnet` network only and sets the redemption dust threshold to the maximum value of the `uint64` type, the redemption timeout to the maximum value of the `uint32` type, and the redemption slashing amount to zero.

### CODE REVIEW

File: solidity/deploy/07_disable_redemptions.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const { execute, read } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  deployments.log("disabling redemptions in the Bridge")
+
+  // To disable redemptions, we need to make the redemption dust threshold
+  // extremely high. The redemption dust threshold value is determined by the
+  // redemptionDustThreshold governance parameter. This parameter is uint64.
+  // The maximum value for the uint64 is 2^64-1 = 18446744073709551615.
+  const redemptionDustThreshold = ethers.BigNumber.from("18446744073709551615")
```
Comment (pdyraga) [inline solidity/deploy/07_disable_redemptions.ts:15]: I do not think this is enough, `2^64-1` in e18 precision is just 18 TBTC. It is not a tiny amount but it's also not that high amount. We need one more thing: what about setting `redemptionTimeout` to the max possible value of `2^32-1 = 4294967295` which is 136 years?

File: solidity/deploy/07_disable_redemptions.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  await execute(
+    "Bridge",
+    { from: deployer, log: true, waitConfirmations: 1 },
+    "updateRedemptionParameters",
+    redemptionDustThreshold,
+    redemptionParameters.redemptionTreasuryFeeDivisor,
+    redemptionParameters.redemptionTxMaxFee,
+    redemptionParameters.redemptionTxMaxTotalFee,
+    redemptionParameters.redemptionTimeout,
+    redemptionParameters.redemptionTimeoutSlashingAmount,
```
Comment (pdyraga) [inline solidity/deploy/07_disable_redemptions.ts:29]: Per _Adjust parameters for Chaosnet phase 1_ section of https://github.com/keep-network/tbtc-v2/issues/438, let's set the slashing amount to 0.

File: solidity/deploy/07_disable_redemptions.ts (hunk 1 above)
Comment (pdyraga) [reply solidity/deploy/07_disable_redemptions.ts:15]: I've just merged https://github.com/keep-network/tbtc-v2/pull/446 to unblock this work, but if we decide to set `redemptionTimeout` to the max value allowed by the type, I think we should do the same for `fraudChallengeDefeatTimeout` in `06_disable_fraud_challenges.ts`, for consistency.

File: solidity/deploy/07_disable_redemptions.ts (hunk 2 above)
Comment (lukasz-zimnoch) [reply solidity/deploy/07_disable_redemptions.ts]: See d372066d

File: solidity/deploy/07_disable_redemptions.ts (hunk 1 above)
Comment (lukasz-zimnoch) [reply solidity/deploy/07_disable_redemptions.ts:15]: See d372066d and f236d3cc

**Review by pdyraga — APPROVED** [review]

**Pull request merged**